dbsnp-importer/src/test/java/uk/ac/ebi/eva/dbsnpimporter/configuration/ImportVariantsJobConfigurationTest.java

```diff
@@ -0,0 +1,224 @@
+@DirtiesContext
+@ContextConfiguration(classes = {ImportVariantsJobConfiguration.class, MongoTestConfiguration.class,
+        JobTestConfiguration.class, EvaRepositoriesConfiguration.class})
+public class ImportVariantsJobConfigurationTest {
+
+    private static final String BATCH_NAME = "CHICKEN_SNPS_BROILER";
+
+    private static final int BATCH_ID = 11825;
+
+    private static final int NON_EXISTENT_BATCH_ID = 1;
+
+    @Autowired
+    private DbsnpTestDatasource dbsnpTestDatasource;
+
+    @Autowired
+    private JobLauncherTestUtils jobLauncherTestUtils;
+
+    @Autowired
+    private JobExplorer jobExplorer;
+
+    @Autowired
+    private Parameters parameters;
+
+    @Autowired
+    private MongoOperations mongoOperations;
+
+    @Autowired
+    private VariantRepository variantRepository;
+
+    // the assembly checker is mocked to avoid adding a large FASTA file to the resources directory
+    @MockBean
+    private AssemblyCheckFilterProcessor assemblyCheckerMock;
+
+    @Before
+    public void setUp() throws Exception {
+        // the assembly checker mock will filter out one variant
+        when(this.assemblyCheckerMock.process(anyObject())).thenAnswer(invocationOnMock -> {
+            SubSnpCoreFields inputVariant = invocationOnMock.getArgumentAt(0, SubSnpCoreFields.class);
+            if (inputVariant.getRsId() == 3136865) {
+                return null;
+            } else {
+                return inputVariant;
+            }
+        });
+    }
+
+    @Test
+    public void loadVariantsAndFile() throws Exception {
+        parameters.setBatchId(BATCH_ID);
+        assertEquals(0, mongoOperations.getCollection(parameters.getVariantsCollection()).count());
+        assertEquals(0, mongoOperations.getCollection(parameters.getFilesCollection()).count());
+
+        JobExecution jobExecution = jobLauncherTestUtils.launchJob();
+
+        assertCompleted(jobExecution);
+
+        DBCollection collection = mongoOperations.getCollection(parameters.getVariantsCollection());
+        List<DBObject> dbObjects = collection.find().toArray();
+        int totalSubsnps = 0;
+        int totalSnps = 0;
+        for (DBObject dbObject : dbObjects) {
+            BasicDBList ids = (BasicDBList) dbObject.get("dbsnpIds");
+            totalSnps += ids.stream().filter(o -> ((String) o).startsWith("rs")).count();
+            totalSubsnps += ids.stream().filter(o -> ((String) o).startsWith("ss")).count();
+        }
+
+        assertEquals(8, dbObjects.size());
+        assertEquals(8, totalSnps);
+        assertEquals(11, totalSubsnps);
+
+        checkASnp();
+        checkAnInsertion();
+
+        checkGenotypes();
+        checkStatistics();
+
+        checkVariantSource();
+    }
+
+    @Test
+    public void doNotLoadBatchIfNoVariants() throws Exception {
+        parameters.setBatchId(NON_EXISTENT_BATCH_ID);
+        assertEquals(0, mongoOperations.getCollection(parameters.getVariantsCollection()).count());
```

Comment:
Those two asserts are in both tests and they would be likely needed in any new test in this class, so I guess they can be moved to the setup method so they are executed before each test